### PR TITLE
Include map file in release build and fix optimizations

### PR DESCRIPTION
**Pull request by ssvine**
There are several issues with release builds:

1. If `NetBox` crashes and `Far` is able to create a crash report, that report won't contain meaningful names of `NetBox` functions. A random user may create an issue attaching such a problematic report. So, it's a good idea to include rather small `.map` file in the binary distribution, because `Far` can use them to convert addresses to meaningful names. (`Far` releases include `.map` files for all executables.)
2. Release build is not fully optimized.

This pull request adds `NetBox.map` file to the release build.

It also fixes and simplifies compiler and linker flags among different build types:

- Remove delay load libraries that are statically linked with `Far` - they are already loaded in the process address space. Also remove `oleaut32.dll` - it is not used in the project
- Remove compiler / linker flags that are duplicated, set by default, redundant, or implied by other flags
- Make `Release`, `RelWithDebInfo`, and `MinSizeRel` build types use (basically) the same flags
- Remove /GS- flag as security measure
- Do not use DEBUG *define* for `RelWithDebInfo`

To summarize: `Release` has maximum speed and the whole program optimizations. `RelWithDebInfo` is the same as `Release`, but also generates debug info and the map file. `MinSizeRel` is the same as `Release` but it creates the smallest file.

### CODE REVIEW

**Comment by ssvine** [comment]
This pull request is closed because it will be recreated as a stacked pull request
